datastream-utils/src/test/java/com/linkedin/datastream/common/TestLogUtils.java

```diff
@@ -31,4 +31,27 @@ public void testLogNumberArrayInRange() throws Exception {
 
     Assert.assertEquals(LogUtils.logNumberArrayInRange(null), "[]");
   }
+
+  @Test
+  public void testLogSummarizedTopicPartitionsMapping() throws Exception {
+    List<String> input = Arrays.asList("t-1", "t-3", "t-6", "t-4", "t-7", "t-12", "t-9", "t-10", "t-8");
+    Assert.assertEquals(LogUtils.logSummarizedTopicPartitionsMapping(input), "t:[1, 3-4, 6-10, 12]");
+
+    input = Arrays.asList("t-3", "t-1", "t-4", "t-6", "t-7", "t-8", "t-9", "t-9", "t-9", "t-10", "t-11", "t-12", "t1-9", "t1-10", "t1-11", "t1-12");
+    Assert.assertEquals(LogUtils.logSummarizedTopicPartitionsMapping(input), "t:[1, 3-4, 6-12], t1:[9-12]");
+    // don't modify the original input list
+    Assert.assertEquals(input.get(0), "t-3");
+
+    input = Arrays.asList("t-1", "t-5", "t-6", "t-4", "t-2", "t-3", "t1-1", "t1-5", "t1-6", "t1-4", "t1-2", "t1-3");
+    Assert.assertEquals(LogUtils.logSummarizedTopicPartitionsMapping(input), "t:[1-6], t1:[1-6]");
```

Comment:
one thing for the test: logSummarizedTopicPartitionsMapping you stream the map on keyset(), I don't think keyset() guarantee ordering, so the topics can get reordered and the string may be correct, but the test can fail

May be: what you can do is, split the output string in test by ", " and then sort the list and compare two list

https://stackoverflow.com/questions/1882762/is-the-java-hashmap-keyset-iteration-order-consistent

Reply:
good point